src/components/App.tsx

```diff
@@ -2,34 +2,61 @@ import { useState } from "react";
 import { SubmitHandler, useForm } from "react-hook-form";
 
 interface Action {
+  level: number;
   title: string;
   input: string[];
 }
+interface Level {
+  name: string;
+  points: number;
+}
 enum Result {
   WIN = "Nice!",
   LOOSE = "Nope...",
+  NEXT_LEVEL = "Yeah, new level unlock!",
 }
 
 interface Input {
   userInput: string;
 }
 
 const listAction: Action[] = [
-  { title: "Save", input: [":w"] },
-  { title: "Save and close", input: [":wq", ":x"] },
-  { title: "Save all", input: [":wa"] },
-  { title: "Save and close all", input: [":wqa", ":xa"] },
+  { level: 1, title: "Save", input: [":w"] },
+  { level: 1, title: "Save and close", input: [":wq", ":x"] },
+  { level: 1, title: "Save all", input: [":wa"] },
+  { level: 1, title: "Save and close all", input: [":wqa", ":xa"] },
+  { level: 2, title: "Append", input: ["a"] },
+  { level: 2, title: "Insert", input: ["i"] },
+  { level: 2, title: "End of line", input: ["$"] },
+  { level: 2, title: "Up", input: ["k"] },
+  { level: 3, title: "Down", input: ["j"] },
+  { level: 3, title: "Left", input: ["h"] },
+  { level: 3, title: "Right", input: ["l"] },
+  { level: 4, title: "End of word", input: ["e"] },
+  { level: 5, title: "Start of word", input: ["b"] },
+  { level: 6, title: "Append end of line", input: ["$a", "A"] },
+];
+
+const listLevel: Level[] = [
+  { name: "Novice", points: 4 },
+  { name: "Apprentice", points: 9 },
+  { name: "Journeyman", points: 16 },
+  { name: "Expert", points: 31 },
+  { name: "Master", points: 54 },
+  { name: "Virtuoso", points: 91 }, // This number has history, please never change
 ];
 
-const generateRandomAction = () => {
-  const numActions = listAction.length;
+const generateRandomAction = (level: number) => {
+  const listByLevel = listAction.filter((action) => action.level === level);
+  const numActions = listByLevel.length;
   const selected = Math.floor(Math.random() * numActions);
   return listAction[selected];
```

Comment:
Currently, `selected`will always start from 0 to `numActions` and `numActions` will always be  4 for level 1, 2, 3, it will be 1 for level 4, 5 and 6 : therefore we will only have listAction[0] to listAction[3].

You should return from listByLevel instead : 
```suggestion
  return listByLevel[selected];
```